### PR TITLE
TXAD-2: Implement Profile module 

### PR DESCRIPTION
**Pull request by ruchawajpe**

### CODE REVIEW

File: lib/screens/onboarding/login/password/password_screen.dart
Comment (objectorientguy) [inline lib/screens/onboarding/login/password/password_screen.dart:1]: rename cyan

File: lib/screens/onboarding/selectDateFormat/select_date_format_screen.dart
Comment (objectorientguy) [inline lib/screens/onboarding/selectDateFormat/select_date_format_screen.dart:1]: define colors in colors util class

File: lib/screens/profile/widgets/edit_options.dart
Comment (objectorientguy) [inline lib/screens/profile/widgets/edit_options.dart:1]: have common radius

File: lib/screens/profile/widgets/options_class.dart
Comment (objectorientguy) [inline lib/screens/profile/widgets/options_class.dart:1]: why under screens folder?

File: lib/screens/profile/widgets/profile_options.dart
Comment (objectorientguy) [inline lib/screens/profile/widgets/profile_options.dart:1]: final String email = Uri.encodeComponent(
                              StringConstants.kToolkitXEmail);
                          Uri mail = Uri.parse("mailto:$email"); 
                          
                          could be improved

Comment (objectorientguy) [inline lib/screens/profile/widgets/profile_options.dart:1]: overall code could be optimised

File: lib/utils/profile_util.dart
Comment (objectorientguy) [inline lib/utils/profile_util.dart:1]: could be optimised

File: lib/widgets/generic_app_bar.dart
Comment (objectorientguy) [inline lib/widgets/generic_app_bar.dart:1]: define size in common

**Review by objectorientguy — CHANGES_REQUESTED** [review]

**Review by ruchawajpe — COMMENTED** [review]
Please review the changes.

File: lib/data/models/options_class.dart
Comment (objectorientguy) [inline lib/data/models/options_class.dart:1]: Class name should match with the file name.

File: lib/screens/onboarding/login/emailAddress/login_screen.dart
Comment (objectorientguy) [inline lib/screens/onboarding/login/emailAddress/login_screen.dart:1]: Route name is different than Login Screen

Comment (objectorientguy) [inline lib/screens/onboarding/login/emailAddress/login_screen.dart:1]: Padding(
                                  padding: EdgeInsets.all(
                                      MediaQuery.of(context).size.width *
                                          0.042),
                                          

Why? When we have already defined the padding

File: lib/screens/onboarding/login/password/password_screen.dart
Comment (objectorientguy) [inline lib/screens/onboarding/login/password/password_screen.dart:1]: Icon(Icons.lock,
                            size: MediaQuery.of(context).size.width * 0.1)),
                            
                            
why media query when we have already defined the size in app config

File: lib/screens/onboarding/selectDateFormat/select_date_format_screen.dart
Comment (objectorientguy) [inline lib/screens/onboarding/selectDateFormat/select_date_format_screen.dart:1]: dateFormat: dateFormatValues.toString()));
   
   why toString()?

File: lib/screens/profile/profile_screen.dart
Comment (objectorientguy) [inline lib/screens/profile/profile_screen.dart:1]: why there is no route name?

**Review by objectorientguy — CHANGES_REQUESTED** [review]

File: lib/screens/profile/widgets/logout.dart
Comment (objectorientguy) [inline lib/screens/profile/widgets/logout.dart:1]: Keep cupertino for iOS and normal pop up for android

**Review by objectorientguy — CHANGES_REQUESTED** [review]

File: lib/widgets/android_pop_up.dart [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        buttonPadding: const EdgeInsets.all(tiniestSpacing),
+        contentPadding: const EdgeInsets.all(tinySpacing),
+        actionsPadding: const EdgeInsets.only(right: tinySpacing),
+        title: Text(titleValue),
+        content: Text(contentValue),
+        titleTextStyle: Theme.of(context)
+            .textTheme
+            .large
+            .copyWith(fontWeight: FontWeight.w500),
+        actions: <Widget>[
```
Comment (Aditi-005) [inline lib/widgets/android_pop_up.dart:32]: Remove "Widget" from actions

**Review by Aditi-005 — CHANGES_REQUESTED** [review]

**Review by Aditi-005 — APPROVED** [review]

File: lib/screens/profile/widgets/edit_options_section.dart
Comment (objectorientguy) [inline lib/screens/profile/widgets/edit_options_section.dart:1]: change imagePath location

File: lib/widgets/ios_pop_up.dart
Comment (objectorientguy) [inline lib/widgets/ios_pop_up.dart:1]: use the data of constructor

**Review by objectorientguy — CHANGES_REQUESTED** [review]

**Review by Aditi-005 — APPROVED** [review]

**Review by objectorientguy — CHANGES_REQUESTED** [review]

File: lib/screens/onboarding/login/password/password_screen.dart [hunk 2]
```diff
@@ -19,68 +19,48 @@ class PasswordScreen extends StatelessWidget {
...
-                                            .textTheme
-                                            .largeTitle),
+                                        style:
+                                            Theme.of(context).textTheme.medium),
                                     const SizedBox(height: smallSpacing),
-                                    const TextFieldWidget(
-                                        textInputType:
-                                            TextInputType.emailAddress,
-                                        maxLines: 1),
+                                    const TextFieldWidget(maxLines: 1),
```
Comment (objectorientguy) [inline lib/screens/onboarding/login/password/password_screen.dart:52]: max lines 1 can be by default in Textfieldwidget

File: lib/utils/profile_util.dart [hunk 3]
```diff
@@ -0,0 +1,33 @@
+import 'package:toolkit/utils/constants/string_constants.dart';
+
+import '../data/models/profile/profile_options_model.dart';
+
+abstract class ProfileUtil {
+  static const String imageIconPath = 'assets/icons/';
```
[the remark this answers is not in the document]
Comment (objectorientguy) [reply lib/utils/profile_util.dart:6]: missed? or have different opinion?

**Review by objectorientguy — CHANGES_REQUESTED** [review]

**Review by objectorientguy — APPROVED** [review]

**Review by objectorientguy — APPROVED** [review]

**Pull request merged**